buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java

```diff
@@ -343,36 +342,36 @@ public void freeThreadLocalCache() {
     }
 
     final class PoolThreadLocalCache extends FastThreadLocal<PoolThreadCache> {
-        private final AtomicInteger index = new AtomicInteger();
-        final AtomicInteger caches = new AtomicInteger();
 
         @Override
-        protected PoolThreadCache initialValue() {
-            caches.incrementAndGet();
-            final int idx = index.getAndIncrement();
-            final PoolArena<byte[]> heapArena;
-            final PoolArena<ByteBuffer> directArena;
-
-            if (heapArenas != null) {
-                heapArena = heapArenas[Math.abs(idx % heapArenas.length)];
-            } else {
-                heapArena = null;
-            }
+        protected synchronized PoolThreadCache initialValue() {
```

Comment:
I think you only need the synchronized of areas are not null and the length > 1. So Maybe do the check of this before synchronize ?